test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -441,7 +562,141 @@ bool VerifyShowToast_Unpackaged()
         return false;
     }
 
-    // TODO: Verify the toast was posted by calling History APIs.
+    return VerifyToastIsActive(toast.ToastId());
+}
+
+bool VerifyRemoveAllAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast3);
+
+    auto getAllAsync = toastNotificationManager.GetAllAsync();
+    if (getAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        getAllAsync.Cancel();
+        return false;
+    }
+
+    // At this point there at least 3 active toasts, the ones we just posted but there could also be others that are still active.
+    if (getAllAsync.get().Size() < 3)
+    {
+        return false;
+    }
+
+    auto removeAllAsync = toastNotificationManager.RemoveAllAsync();
+    if (removeAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        removeAllAsync.Cancel();
+        return false;
+    }
+
+    getAllAsync = toastNotificationManager.GetAllAsync();
+    if (getAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Error)
+    {
+        getAllAsync.Cancel();
+        return false;
+    }
+
+    return true;
+}
+
+bool VerifyGetAllAsyncWithZeroActiveToast()
+{
+    EnsureNoActiveToasts();
+
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    auto getAllAsync = toastNotificationManager.GetAllAsync();
+    if (getAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Error)
+    {
+        getAllAsync.Cancel();
+        return false;
+    }
+
+    return true;
+}
+
+bool VerifyGetAllAsyncWithOneActiveToast()
+{
+    EnsureNoActiveToasts();
+
+    winrt::ToastNotification toast{ CreateToastNotification(L"MyOwnLittleToast")};
+    toast.Tag(L"aDifferentTag");
+    toast.Group(L"aDifferentGroup");
+    winrt::DateTime expirationTime{ winrt::clock::now() };
+    expirationTime += winrt::TimeSpan{ std::chrono::seconds(10) };
+    toast.ExpirationTime(expirationTime);
+    toast.ExpiresOnReboot(false); // ELx - Setting this to true fails, investigating
+    toast.Priority(winrt::Microsoft::Windows::ToastNotifications::ToastPriority::High);
```

Comment:
It can be summarized as:

```suggestion
    toast.Priority(winrt::ToastPriority::High);
```

Reply:
Done